src/lib/mdns/Discovery_ImplPlatform.cpp

```diff
@@ -435,6 +435,14 @@ CHIP_ERROR DiscoveryImplPlatform::ResolveNodeId(const PeerId & peerId, Inet::IPA
 void DiscoveryImplPlatform::HandleCommissionableNodeBrowse(void * context, MdnsService * services, size_t servicesSize,
                                                            CHIP_ERROR error)
 {
+    if (servicesSize == 0 || services == nullptr || error != CHIP_NO_ERROR)
```

Comment:
That is because if you request DNS browse and no service will be found, OnNodeDiscoeryComplete will not be called. So basically you don't get any information that timeout expired and what has happened.